src/v3/src/transformer/selectAuthenticator/transformSelectAuthenticatorEnroll.test.ts

```diff
@@ -190,4 +213,288 @@ describe('Enroll Authenticator Selector Transformer Tests', () => {
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.buttons[0] as AuthenticatorButtonElement).label).toBe('Email');
+  });
+
+  it('should transform authenticator elements when all elements have expired grace periods', () => {
+    const options = [
+      {
+        relatesTo: {
+          gracePeriod: {
+            expiry: '2020-09-27 18:00:00.000.',
+          },
+        },
+      } as unknown as IdxOption,
+      {
+        relatesTo: {
+          gracePeriod: {
+            expiry: '2020-9-27 18:00:00.000.',
+          },
+        },
+      } as unknown as IdxOption,
+    ];
+
+    transaction.nextStep = {
+      name: IDX_STEP.SELECT_AUTHENTICATOR_ENROLL,
+      canSkip: isSkippable.mockReturnValue(false)(),
+      inputs: [{
+        name: 'authenticator',
+        options,
+      }],
+    };
+
+    transaction.availableSteps = [{ name: 'skip', action: jest.fn() }];
+
+    const updatedFormBag = transformSelectAuthenticatorEnroll({
+      transaction, formBag, widgetProps,
+    });
+
+    expect(updatedFormBag).toMatchSnapshot();
+    expect(mockGetAuthenticatorEnrollButtonElementsFn).toBeCalledTimes(2);
+    expect(mockGetAuthenticatorEnrollButtonElementsFn).toBeCalledWith([], 'select-authenticator-enroll', 'ok_pl', undefined);
+    expect(mockGetAuthenticatorEnrollButtonElementsFn).toBeCalledWith(options, 'select-authenticator-enroll', 'ok_pl', undefined);
+    expect(updatedFormBag.uischema.elements.length).toBe(5);
+    expect(updatedFormBag.uischema.elements[0].type).toBe('Title');
+    // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+    expect(updatedFormBag.uischema.elements[0].options.content).toBe('oie.select.authenticators.enroll.title');
+    expect(updatedFormBag.uischema.elements[1].type).toBe('Description');
+    // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+    expect(updatedFormBag.uischema.elements[1].options.content).toBe('oie.select.authenticators.enroll.subtitle');
+    expect(updatedFormBag.uischema.elements[2].type).toBe('Heading');
+    // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+    expect(updatedFormBag.uischema.elements[2].options.content).toBe('oie.setup.optional.short');
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.dataSe)).toBe('authenticator-enroll-list');
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.buttons[0] as AuthenticatorButtonElement).options.type).toBe(ButtonType.BUTTON);
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.buttons[0] as AuthenticatorButtonElement).label).toBe('Email');
+    expect((updatedFormBag.uischema.elements[4] as ButtonElement).options.step)
+      .toBe('skip');
```

Comment:
If all grace periods expired, I wouldn't expect there to be a `skip` element. Am I reading this test incorrectly?

Reply:
It seems like the first new test should have the `skip` step included, but this one wouldn't get it

Reply:
Discussed with @jakebacher-okta , this is intended so that we test that we're reactive to the backend response i.e. skip in response should skip button in the UI. But no, it's not likely that this is a possible/valid idx response to have the skip + expired grace periods.

Reply:
Understood. I think I'm also confused by the assertion that `oie.setup.optional.short` string is shown even in the case where no authenticators have grace period. I understand that if `skip` is available, then the description being "optional" is consistent, but in terms of realistic scenarios, it is a bit misleading to any reader of the test. If there's a way to move some of the scenarios around to be more realistic that would be best